### PR TITLE
Keep yaw setpoint within +/-delta from the current yaw

### PR DESCRIPTION
**Pull request by matejkarasek**
Adding an optional check for the distance of the yaw setpoint from the current heading (yaw).
If the difference is larger than +/- YAW_MAX_DELTA, the setpoint is kept within the +/- YAW_MAX_DELTA interval.

This functionality is useful for drones with limited yaw control authority.
It is turned off by default and can be enabled by defining YAW_MAX_DELTA in config.mk

### CODE REVIEW

**Comment by matejkarasek** [comment]
Any thoughts on this PR?

Two cases when this is useful:
1) A human pilot is giving a constant yaw rate setpoint, which the aircraft cannot follow (not enough control authority, or because of a disturbance). Thus the error between the yaw angle setpoint (integrated yaw rate setpoint) and the current yaw keeps increasing. Once the pilot wants to stop the rotation, and commands zero yaw rate, the aircraft will still keep rotating until the setpoint is reached (which can be another 90 degrees or more). 
The YAW_MAX_DELTA parameter allows to limit this behavior.

2) Same situation as in 1), the error keeps growing. When it goes over 180 degrees, **the aircraft will suddenly start rotating in the opposite direction** (as it is "shorter" this way to reach the angular setpoint).
The YAW_MAX_DELTA prevents this undesirable change of sense of rotation.

And another case where it could help, if extended also to modeAbsolute (now not implemented)
3) If the yaw setpoint is changed by 180 degrees (e.g. by the commander) AND there is a lot of noise in the yaw estimation, the controler might "hesitate" about the direction in which the aircraft should rotate (one sample clockwise is shorter, another CCW might be shorter). The YAW_MAX_DELTA would prevent this.

**Comment by matejkarasek** [comment]
Thanks for the feedback! Will test it out end of next week.

File: src/modules/src/controller_pid.c [hunk 1]
```diff
@@ -65,7 +65,20 @@ void controllerPid(control_t *control, setpoint_t *setpoint,
   if (RATE_DO_EXECUTE(ATTITUDE_RATE, tick)) {
     // Rate-controled YAW is moving YAW angle setpoint
     if (setpoint->mode.yaw == modeVelocity) {
-       attitudeDesired.yaw += setpoint->attitudeRate.yaw * ATTITUDE_UPDATE_DT;
+      attitudeDesired.yaw = capAngle(attitudeDesired.yaw + setpoint->attitudeRate.yaw * ATTITUDE_UPDATE_DT);
```
[the remark this answers is not in the document]
Comment (matejkarasek) [reply src/modules/src/controller_pid.c:68]: This should not be a problem, any discontinuity in the yaw angle error is already handled here:
https://github.com/flapper-drones/crazyflie-firmware/blob/5534c8e4e26b7955588fde40d8f422db464957bc/src/modules/src/attitude_pid_controller.c#L128

**Comment by matejkarasek** [comment]
So I tested this on a Crazyflie, continuously yawing left and right, no problems observed

If little amount of throttle is commanded, such that the props spin but the CF stays on the ground, the effect of the YAW_MAX_DELTA can be clearly observed:
- without the define, when commanding constant yaw rate, the crazyflie alternates between left and right yawing
- with the define, when commanding constant yaw rate, the crazyflie keeps trying to yaw in only the correct direction

**Comment by knmcguire** [comment]
Yeah no objections here either:) Merging!

**Pull request merged**